src/api/spec/models/obs_factory/openqa_job_spec.rb

```diff
@@ -0,0 +1,165 @@
+require 'rails_helper'
+require 'webmock/rspec'
+
+RSpec.describe ObsFactory::OpenqaJob do
+  describe '.openqa_base_url' do
+    it { expect(ObsFactory::OpenqaJob.openqa_base_url).to eq('http://openqa.opensuse.org') }
+
+    context 'with a configured base_url' do
+      before do
+        CONFIG['openqa_base_url'] = 'http://configured-url.com'
+      end
+
+      it { expect(ObsFactory::OpenqaJob.openqa_base_url).to eq('http://configured-url.com') }
+    end
+  end
+
+  describe '.openqa_links_url' do
+    it { expect(ObsFactory::OpenqaJob.openqa_links_url).to eq('https://openqa.opensuse.org') }
+
+    context 'with a configured links_url' do
+      before do
+        CONFIG['openqa_links_url'] = 'http://links-url.com'
+      end
+
+      it { expect(ObsFactory::OpenqaJob.openqa_links_url).to eq('http://links-url.com') }
+    end
+  end
+
+  describe '#result_or_state' do
+    subject { ObsFactory::OpenqaJob.new(result: 'some-result', state: 'some-state') }
+
+    it { expect(subject.result_or_state).to eq('some-result') }
+
+    context 'with a result value of "none"' do
+      subject { ObsFactory::OpenqaJob.new(result: 'none', state: 'some-state') }
+
+      it { expect(subject.result_or_state).to eq('some-state') }
+    end
+  end
+
+  describe '#failing_modules' do
+    let(:modules) do
+      %w[passed softfailed running none failed].map do |result|
+        { 'name' => "mod_#{result}", 'result' => result }
+      end
+    end
+
+    subject { ObsFactory::OpenqaJob.new(modules: modules) }
+
+    it { expect(subject.failing_modules).to contain_exactly('mod_failed') }
```

Comment:
I would at least put a label to this it block. Right now it seems a bit magical what this method does.